Adds namespacing and makes function naming more consistent.

batesweb opened the pull request:
Closes #769

### DESCRIPTION

Removes the function prefixes (`_s`) and adds the `WD_S` namespace throughout the theme.
Updates function names to be more consistent. Functions that render markup start with `display_` and functions that return data start with `return_`.

### SCREENSHOTS

Doesn't change anything visual in the theme. Just refactors the existing code.

### OTHER

- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT?

### STEPS TO VERIFY

Test all theme functionality to make sure there are no namespacing issues.

### DOCUMENTATION

Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

batesweb committed:
Reorganizes the template tags functions

batesweb committed:
Breaks all of the functions in inc/ files out into individual files

batesweb committed:
Replaced underscores with hyphens in filenames

batesweb committed:
Added doc block comments

batesweb committed:
Added readme to inc/

batesweb committed:
Working on adding namespaces...

batesweb committed:
More namespacing updates....

batesweb committed:
More namespacing updates...

batesweb committed:
Fix SVG namespacing

batesweb committed:
Fixes function comments

batesweb committed:
Adds .gitkeep file to force creation of empty post-types directory

batesweb committed:
Changed function namespace imports

batesweb committed:
Fixed page template namespacing

jrfoell commented on inc/setup/preload-assets.php:
```diff
@@ -0,0 +1,24 @@
+function preload_assets() {
+	?>
+	<?php if ( return_custom_logo_url() ) : ?>
+		<link rel="preload" href="<?php echo esc_url( Functions\return_custom_logo_url() ); ?>" as="image">
```
I think this can just be `return_custom_logo_url()`

batesweb replied:
Good catch!

jrfoell reviewed (commented):
I like where this is going! I think we need to discuss the actual namespace prefix(es) but that can be a group discussion :sunglasses:

batesweb committed:
Added namespace magic constant

batesweb committed:
Fixed return_custom_logo_url namespace

batesweb committed:
Missed a few __NAMESPACE__ constants

aubreypwd commented on inc/template-tags/display-svg.php:
```diff
@@ -0,0 +1,58 @@
+ * @package _s
+ */
+
+namespace WD_S\TemplateTags;
```
I am in favor of a different format for namespaces across all files:

```
namespace wd_s\template_tags;
namespace wd_s\functions;
```

..etc

I think that using PSR4-ish flavoring here is unwarranted as we are only using namespacing here to avoid prefixing and avoid function collisions. I think going with _lowercase_ (since everything else is) and using `_`'s instead of `PascalCase`  keeps things nice and consistent.

aubreypwd replied:
I also would prefer using `webdevstudios\` in the namespace:

```
namespace webdevstudios\wd_s\template_tags
namespace webdevstudios\wd_s\functions
```

And I would expect all our projects to follow this format.

aubreypwd replied:
@salcode I know we somewhat shared opinions on this? What do you think?

salcode replied:
My primary thought is I don't want @batesweb to have to keep modifying this PR 😆 , perhaps we want to have a small group call to come to a consensus on direction.

Personally:
1. I agree with preferring the full company name for the namespace. I think it is good for our brand and more inline with other namespaces I see being used.
2.  I'd prefer PascalCase, `WebDevStudios`, rather than all lower-case `webdevstudios`). I like PSR-4ish namespaces.
3. Using sub-namespaces `WebDevStudios\functions` and `WebDevStudios\TemplateTags` is an interesting idea that I'm open to but not locked in on in terms of an opinion
4. For function name prefixes, I'd much prefer `get_` prefixed functions rather than `return_`

Related namespace reading https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/ (TLDR: This advocates for Camel_Caps for namespaces, e.g. `My_Awesome_Plugin\Admin`, which I think would still be `WebDevStudios` since it is one word).

aubreypwd replied:
I would totally be OK with `PascalCase` as long as we use `_`'s e.g. `\WebDevStudios\Functions\Pascal_Case`, etc :D

salcode replied:
After giving this more thought, I think we should use the same namespace, `WebDevStudios\wd_s`, on all files.

i.e.

```
namespace WebDevStudios\wd_s;
```

1. `{organization}\{product}` is a common format for namespaces, in this case our product is `wd_s`
2. I like the all lower-case `wd_s` because, based on the [README.md](https://github.com/WebDevStudios/wd_s#readme), the name of this product _is all lowercase_
3. Using one namespace for the entire project
    - Advantage: we don't have to worry about what namespace a function is in, they are all in the same one
    - Advantage: we don't need any `use function` lines because we don't need to import a function from another namespace
    - Disadvantage: we can't re-use the same function name in two different places (e.g. in `functions` and in `TemplateTags`) - I don't think we should be doing this anyway
    - Advantage: If we want to change the namespace (e.g. a client doesn't want our branding in the namespace) it is a relatively simple search and replace across all of the files
    - Note: when we do need to use the namespace (e.g. in a `add_action()` or `add_filter()`, see examples in blog post [WordPress Hooks and PHP Namespaces](https://salferrarello.com/wordpress-hooks-and-php-namespaces/)), we should always use `__NAMESPACE__` to maintain the ability to change the namespace

aubreypwd replied:
> I think we should use the same namespace, `WebDevStudios\wd_s`, on all files [...] Disadvantage: we can't re-use the same function name in two different places

I think expanding on the namespacing ensures comfort and ease knowing you will have less function name conflicts, e.g. I can have a `get_my_slug()` in two different context/files vs having to prefix it. Hence why I would disagree here. 

But not strongly, but I think being able to expand on Namespaces should be allowed and encouraged. I think either way we go is going to be _better_! That being said, I personally like the direction @batesweb is going here, and I would keep it e.g. `WebDevStudios\wd_s\Template_Tags`, `WebDevStudios\wd_s\Functions`, etc.

salcode replied:
I brought this topic to the larger BEE group at the 2022-01-06 BEE Scrum and the general consensus was to use a single namespace across all files of

```
namespace WebDevStudios\wd_s;
```

jrfoell replied:
To add context:

* There are currently no classes in `wd_s` so the namespace will not be used for directory/path autoloading
* The purpose of adding namespaces was to eliminate global function prefixing ala `function _s_something()`
* By having one namespace, we'll never have to guess which namespace we're in and all hooks can simply use the `__NAMESPACE__` constant

jrfoell replied:
The BEE team discussed `return_` && `display_` (vs. WP's `get_` && `the_`).

We unanimously agreed that `get_` should be kept instead of `return_` because it just makes sense and with the namespace added, there should be no confusion where the function lives.

jrfoell replied:
BEEs like `print_` better than `display_` and agree that `the_` is just wrong :stuck_out_tongue:

aubreypwd reviewed (commented):
I can neither approve or reject this, I think either way we go Namespacing is super helpful here. I also like the nested namespacing e.g. for `wd_s\functions` and `wd_s\template_tags` vs. just piling them all in a single `wd_s` namespace.

Please see my advise though, as I think we can format this a little better...

coreymcollins commented:
@batesweb – it looks like this branch is a bit out of date now, and I'm not sure the full status of these updates. When you get a chance, can you review this branch, bring it up to date, and set it R4R if it's ready for final testing & review? Thanks!

nickyiie committed:
changed namespace to WebDevStudios/wd_s

nickyiie committed:
changed namespace

nickyiie committed:
renaming of functions/files

nickyiie committed:
fixed typo on __return_false

nickyiie committed:
pulling main

nickyiie commented:
@coreymcollins 

branch has been updated and includes all final naming conventions as described by @jrfoell in the discussion above.

coreymcollins committed:
Fixes some JS linting issues

coreymcollins committed:
Replaces old namespacing

batesweb committed:
Use single namespace throughout theme

batesweb committed:
Renames theme get_the_title function

coreymcollins commented:
One of the last things we'll need for this, beyond our reviews, is documentation that we can add to the wiki. This includes updating the existing pages where functions are being referenced.

@batesweb, would you be able to take that on?

batesweb committed:
Cleans up namespacing in template tags

batesweb committed:
More function and namespace updates...

batesweb committed:
Renames the title and excerpt functions

oliverharrison committed:
Merge branch 'main' into feature/#769-function-naming

oliverharrison committed:
Updated to use either namespaces or functions as appropriate

oliverharrison committed:
Updated package lock

oliverharrison committed:
Add 'use function' instead of declaring the namespace again; remove namespace declaration where unnecessary

oliverharrison committed:
Change 'get_the_content' function to 'get_post_content' to avoid conflicts with built-in function name

asharirfan committed:
Add missing namespaces

asharirfan committed:
Fix spacing issues

asharirfan committed:
Remove namespace from backward compatibility function

Removing namespace from this file because this function needs to be in WP's namespace if it is not defined for backwards compatibility.

asharirfan committed:
Add missing nav function for scaffolding

asharirfan committed:
Add missing namespace

asharirfan committed:
Fix attribute escape function

asharirfan committed:
Fix spacing issues

asharirfan committed:
Remove non-required namespace function declarations

asharirfan committed:
Add namespace function declaration

asharirfan committed:
Fix namespace declaration

asharirfan commented:
@oliverharrison @batesweb I have made some changes to the PR with regards to how the namespace is being used.

So the general rule of thumb which I follow is that we declare namespace in all of the function definition files like the following. This makes the function part of the namespace space that we are declaring.

```
namespace WebDevStudios\wd_s;
```

And whenever this statement is put on top of any file, we don't need to add the `use function WebDevStudios\wd_s\<function-name>` statement because our function already exists in that namespace. We only add the `use` statement to add a function from a difference namespace.

So, for example, in [content.php](https://github.com/WebDevStudios/wd_s/blob/feature/%23769-function-naming/template-parts/content.php#L10) since we are not defining our namespace there, we need to add the `use function` statement at the top — which makes the functions available in this file.

Please let me know if you have any questions or confusions. We can schedule a call if needed.

oliverharrison committed:
Update text domains and package names

oliverharrison committed:
Update text domain

oliverharrison committed:
Update spacing

oliverharrison committed:
Update prettier config to handle wp-scripts format correctly

jrfoell reviewed (approved):
Putting my stamp of approval on this for the BEE team at large b/c I think this hits the mark of what we were suggesting :muscle:

oliverharrison committed:
Fix scaffolding expand/collapse sections

oliverharrison committed:
Fix scaffolding nav

oliverharrison committed:
Update phpcs rule to capture namespacing

oliverharrison committed:
Update handle prefixes

oliverharrison committed:
update _s.pot -> wd_s.pot for consistency

batesweb committed:
Removed the old post_date function AGAIN

batesweb committed:
Updates excerpt function name

oliverharrison committed:
Include all files with a function using namespacing

oliverharrison committed:
Define function prefix for functions in global namespace

batesweb committed:
Cleans up prefixes

oliverharrison committed:
Update function and filename for clarity

oliverharrison committed:
Update _wd_s_ -> wd_s_

jrfoell reviewed (approved):
Liking these latest changes using `print_` and `get_` from the BEE feedback.

Commit:
Fixes ability to save content in Copyright field in theme customizer

Commit:
Fixes functions for printing customizer header and footer scripts to use the custom function (get_post_content) to get content instead of wordpress default get_the_content

Commit:
Updates comments about find and replace to refer to readme file.

Commit:
Updates readme with complete find and replace list

oliverharrison committed:
Update tested up to version

jrfoell commented on header.php:
```diff
@@ -26,9 +26,9 @@
 <body <?php body_class( 'site-wrapper' ); ?>>
 
-	<?php wp_body_open(); ?>
+	<?php wd_s_wp_body_open(); ?>
```
Noting that we should keep `wp_body_open()` and instead remove the (weirdly named) `wd_s_wp_body_open()` as this theme already specifies a minimum WP version of 5.6 so the backwards compatibility is unnecessary. See https://generatewp.com/wordpress-5-2-action-that-every-theme-should-use/

oliverharrison replied:
Sounds good to me.

jrfoell commented on header.php:
```diff
@@ -46,21 +46,21 @@
-				$description = get_bloginfo( 'description', 'display' );
-				if ( $description || is_customize_preview() ) :
+				$wd_s_description = get_bloginfo( 'description', 'display' );
+				if ( $wd_s_description || is_customize_preview() ) :
 					?>
-					<p class="site-description"><?php echo esc_html( $description ); ?></p>
+					<p class="site-description"><?php echo esc_html( $wd_s_description ); ?></p>
```
This seems to have been caught by an overzealous search & replace.

jrfoell replied:
Nevermind, this is best practice to prefix global variables.

jrfoell commented on inc/customizer/customizer.php:
```diff
@@ -2,47 +2,49 @@
 	$wp_customize->remove_section( 'colors' );
 }
-add_action( 'customize_register', '_s_remove_default_customizer_sections', 15 );
+add_action( 'customize_register', __NAMESPACE__ . '\remove_default_customizer_sections', 15 );
```
:heart_eyes:

jrfoell commented on inc/functions/compat.php:
```diff
@@ -0,0 +1,26 @@
+<?php
```
Remove this file since wd_s specifies WP 5.6 and above.

jrfoell committed:
Remove backwards-compat wp_body_open() since wd_s specifies WP 5.6+

jrfoell committed:
Merge branch 'main' into feature/#769-function-naming

asharirfan committed:
Fix script handle prefix

oliverharrison committed:
Update version from 2.1.0 -> 3.0.0

oliverharrison reviewed (approved):
Looks good!

oliverharrison committed:
Re-add version to package.json
